### PR TITLE
Updated documentation

### PR DESCRIPTION
**Pull request by rbrtbnfgl**
## Description

Updated documentations:
- Removed reference to unsupported project
- Updated link to external projects
- Added reference to K3s and RKE2
- Added K3s as Kubernetes example with Flannel
- Moved k8s-manifest to k8s-old-manifest since they are used on older K8s versions
- Moved Wireguard from experimental backend as a recommended one

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Review by thomasferrandiz — APPROVED** [review]

File: Documentation/integrations.md [hunk 1]
```diff
@@ -6,8 +6,8 @@ This document tracks projects that integrate with flannel. [Join the community](
 
 [Kubernetes](https://kubernetes.io/docs/admin/networking/#flannel): Container orchestration platform with options for [flannel as an overlay](https://kubernetes.io/docs/admin/networking/#flannel).
 
-[Tectonic](https://coreos.com/tectonic/): CoreOS's enterprise ready, 100% pure-upstream distribution of Kubernetes.
+[Canal](https://projectcalico.docs.tigera.io/getting-started/kubernetes/flannel/flannel): Policy-based networking for cloud-native apps.
```
Comment (manuelbuil) [inline Documentation/integrations.md:9]: This definition is weird. I would say: Kubernetes CNI plugin that uses Calico for network policies and intra-node communications and Flannel for inter-node communications.

File: Documentation/backends.md [hunk 2]
```diff
@@ -29,6 +29,30 @@ host-gw provides good performance, with few dependencies, and easy set up.
...
+      the peer addresses
+    * ipv6 - Single wireguard tunnel for both address families; use ipv6 for
+      the peer addresses
+* `PersistentKeepaliveInterval` (int): Optional. Default is 0 (disabled).
+
+If no private key was generated before the private key is written to `/run/flannel/wgkey`. You can use environment `WIREGUARD_KEY_FILE` to change this path.
+
+The static names of the interfaces are `flannel-wg` and `flannel-wg-v6`. WireGuard tools like `wg show` can be used to debug interfaces and peers.
+
+Users of kernels < 5.6 need to [install](https://www.wireguard.com/install/) a module.
```
Comment (manuelbuil) [inline Documentation/backends.md:54]: Could you specify the module?

Comment (rbrtbnfgl) [reply Documentation/backends.md:54]: Following the URL you should have what you need to install based on your OS. Ubuntu, debian and openSUSE have different packages to install.

Comment (manuelbuil) [reply Documentation/backends.md:54]: Aaah ok! I think the word module is confusing. What about modifying the `a module` by `an additional wireguard package`

**Review by manuelbuil — APPROVED** [review]

**Pull request merged**